pyanaconda/modules/base.py

```diff
@@ -98,3 +100,56 @@ def publish_task(self, implementation, module_path):
     def ping(self, s):
         """Ping the module."""
         return s
+
+    @property
+    def kickstart_specification(self):
+        """Return a kickstart specification.
+
+        Every kickstart module that is interested in processing
+        kickstart files, should provide its own specification.
+
+        :return: a subclass of KickstartSpecification
+        """
+        return NoKickstartSpecification
+
+    @property
+    def kickstart_command_names(self):
+        """Return a list of kickstart command names."""
+        return list(self.kickstart_specification.commands.keys())
+
+    @property
+    def kickstart_section_names(self):
+        """Return a list of kickstart section names."""
+        return list(self.kickstart_specification.sections.keys())
+
+    def read_kickstart(self, s):
+        """Read the given kickstart string.
+
+        The kickstart string should contain only commands and
+        sections that are defined by the kickstart specification.
+
+        :param s: a kickstart string
+        """
+        spec = self.kickstart_specification
+        handler = get_kickstart_handler(spec)
+        parser = get_kickstart_parser(handler, spec)
+
+        parser.readKickstartFromString(s)
+        self.process_kickstart(handler)
+
+    def process_kickstart(self, data):
+        """Process the kickstart data.
+
+        :param data: a kickstart handler defined by the specification
+        """
+        pass
+
+    def write_kickstart(self):
+        """Return a kickstart representation of this module.
+
+        The kickstart string should contain only commands and
+        sections that are defined by the kickstart specification.
+
+        :return: a kickstart string
+        """
+        return ""
```

Comment:
I'm not sure if this should be part of the Base module class. Logically it seems to me completely else than the publish_task() method. I would rather promote this to a separate class, maybe it would be better to don't have this class in the inheritance of KickstartModule at all.

Reply:
Might be fixed in the future.